File: Daily Test/C/Engulfed Matrix 17.7.2025.c
#include <stdio.h>

int main() {
    int r, c;
    scanf("%d %d", &r, &c);
    int m[r][c];
    for (int i = 0; i < r; i++)
        for (int j = 0; j < c; j++)
            scanf("%d", &m[i][j]);
    int a, b, x, y;
    scanf("%d %d", &a, &b);
    scanf("%d %d", &x, &y);
    int rowStart = a < b ? a : b;
    int rowEnd = a > b ? a : b;
    int colStart = x < y ? x : y;
    int colEnd = x > y ? x : y;

    for (int i = rowStart; i < rowEnd - 1; i++) {
        for (int j = colStart; j < colEnd - 1; j++)
            printf("%d ", m[i][j]);
        printf("\n");
    }

    return 0;
}

File: Daily Test/Java/Engulfed Matrix 17.7.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int r = sc.nextInt(), c = sc.nextInt();
        int[][] m = new int[r][c];
        for (int i = 0; i < r; i++)
            for (int j = 0; j < c; j++)
                m[i][j] = sc.nextInt();
        int a = sc.nextInt(), b = sc.nextInt();
        int x = sc.nextInt(), y = sc.nextInt();

        int rowStart = Math.min(a, b), rowEnd = Math.max(a, b);
        int colStart = Math.min(x, y), colEnd = Math.max(x, y);

        for (int i = rowStart; i < rowEnd - 1; i++) {
            for (int j = colStart; j < colEnd - 1; j++)
                System.out.print(m[i][j] + " ");
            System.out.println();
        }
    }
}